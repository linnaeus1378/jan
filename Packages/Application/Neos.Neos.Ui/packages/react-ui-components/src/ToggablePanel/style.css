.panel {
    composes: reset from './../reset.css';
    position: relative;
}

.panel__headline {
    composes: reset from './../reset.css';
    cursor: pointer;
    height: var(--spacing-GoldenUnit);
    margin: 0;
    padding: 0 var(--spacing-Full);
    line-height: var(--spacing-GoldenUnit);
    font-weight: bold;
    font-size: var(--fontSize-Base);
}

.panel__headline--noPadding {
    padding: 0;
}

.panel__toggleBtn {
    composes: reset from './../reset.css';
    position: absolute;
    top: 0;
    right: 0;
}
.panel__contents {
    composes: reset from './../reset.css';
    padding: 0 var(--spacing-Full);
    transition: var(--transition-Default) ease padding;

    .panel--isOpen & {
        padding-top: 5px;
        padding-bottom: 5px;
    }
    .panel--condensed.panel--isOpen & {
        padding-top: 0;
        padding-bottom: 0;
    }
}

.panel__contents--noPadding {
    padding: 0;
}
