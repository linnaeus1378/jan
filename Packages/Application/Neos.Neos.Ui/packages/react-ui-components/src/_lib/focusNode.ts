/**
 * Creates a curried function which can be used as a `ref` in React Components
 * to focus the rendered node.
 *
 * @param {Boolean} Indicates wether the node should be focused or not,
 */
export const makeFocusNode = (isFocused: boolean) => (node: any) => {
    if (node) {
        const method = isFocused ? 'focus' : 'blur';

        node[method]();
    }
};
