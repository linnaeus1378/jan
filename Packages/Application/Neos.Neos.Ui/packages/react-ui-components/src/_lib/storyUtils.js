import React from 'react';
import PropTypes from 'prop-types';

/**
 * The StoryWrapper is a simple component which can be used to wrap stories
 * of the react-storybook lib.
 */
export const StoryWrapper = ({children, ...rest}) => (
    <div
        style={{background: '#000', padding: '1rem', position: 'relative'}}
        {...rest}
        >
        {children}
    </div>
);
StoryWrapper.propTypes = {
    children: PropTypes.node.isRequired
};
