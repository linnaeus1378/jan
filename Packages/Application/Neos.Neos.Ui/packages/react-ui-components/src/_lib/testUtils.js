import React from 'react';

/**
 * A function which returns a simple React element which can be
 * used as a test stub for cross component dependencies.
 */
export const createStubComponent = componentName => props => <div data-component-name={componentName} {...props} style={{}}/>;
