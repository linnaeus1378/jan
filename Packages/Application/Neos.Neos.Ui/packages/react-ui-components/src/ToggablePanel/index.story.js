import React from 'react';
import {storiesOf, action} from '@storybook/react';
import {withKnobs, boolean} from '@storybook/addon-knobs';
import {StoryWrapper} from './../_lib/storyUtils';
import ToggablePanel from '.';

storiesOf('ToggablePanel', module)
    .addDecorator(withKnobs)
    .addWithInfo(
        'stateful',
        'ToggablePanel - Stateful',
        () => (
            <StoryWrapper>
                <ToggablePanel isOpen={boolean('Is Open?', true)}>
                    <ToggablePanel.Header>
                        Header
                    </ToggablePanel.Header>
                    <ToggablePanel.Contents>
                        Contents
                    </ToggablePanel.Contents>
                </ToggablePanel>
            </StoryWrapper>
        ),
        {inline: true}
    )
    .addWithInfo(
        'stateful - closing towards bottom',
        'ToggablePanel - Closing towards bottom',
        () => (
            <StoryWrapper>
                <div style={{height: '200px', display: 'flex', flexDirection: 'column'}}>
                    <div style={{flexBasis: '100%', color: 'white'}}>
                        Upper container, to push down ToggablePanel
                    </div>
                    <ToggablePanel isOpen={boolean('Is Open?', true)} closesToBottom={boolean('Closes towards bottom?', true)}>
                        <ToggablePanel.Header>
                            Header
                        </ToggablePanel.Header>
                        <ToggablePanel.Contents>
                            Contents
                        </ToggablePanel.Contents>
                    </ToggablePanel>
                </div>
            </StoryWrapper>
        ),
        {inline: true}
    )
    .addWithInfo(
        'stateless',
        'ToggablePanel - Stateless',
        () => (
            <StoryWrapper>
                <ToggablePanel isOpen={boolean('Is Open?', true)} onTogglePanel={action('toggle')}>
                    <ToggablePanel.Header>
                        Header
                    </ToggablePanel.Header>
                    <ToggablePanel.Contents>
                        Contents
                    </ToggablePanel.Contents>
                </ToggablePanel>
            </StoryWrapper>
        ),
        {inline: true}
    );
