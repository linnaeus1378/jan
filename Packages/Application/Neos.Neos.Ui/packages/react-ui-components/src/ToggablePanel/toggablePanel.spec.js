import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {createStubComponent} from './../_lib/testUtils';
import ToggablePanel, {
    StatelessToggablePanel,
    Header,
    Contents
} from './toggablePanel';

describe('<ToggablePanel/>', () => {
    let props;

    beforeEach(() => {
        props = {
            theme: {
                'panel': 'baseClassName',
                'panel--isOpen': 'isOpenClassName',
                'panel__headline': 'panelHeadlineClassName',
                'panel__contents': 'panelContentsClassName'
            },
            HeadlineComponent: createStubComponent('HeadlineComponent'),
            IconButtonComponent: createStubComponent('IconButtonComponent')
        };
        props.children = [
            <Header key="header" {...props}>My Header</Header>,
            <Contents key="contents" {...props}>My my Content</Contents>
        ];
    });

    it('should render correctly.', () => {
        const wrapper = shallow(<ToggablePanel {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<ToggablePanel {...props} className="fooClassName"/>);

        expect(wrapper.prop('className')).toContain('fooClassName');
    });

    it('should allow the propagation of additional props to the wrapper.', () => {
        const wrapper = shallow(<ToggablePanel {...props} foo="bar"/>);

        expect(wrapper.prop('foo')).toBe('bar');
    });

    it('should initialize with a state of {isOpen: props.isOpen}.', () => {
        const wrapper = shallow(<ToggablePanel {...props}/>);

        expect(wrapper.state('isOpen')).toBe(true);
    });

    it('should toggle the "isOpen" state when calling the instance "toggle" method.', () => {
        const wrapper = shallow(<ToggablePanel {...props}/>);

        expect(wrapper.state('isOpen')).toBe(true);

        wrapper.instance().toggle();

        expect(wrapper.state('isOpen')).toBe(false);
    });

    it('should propagate the "isOpen" prop to the the "StatelessToggablePanel" component in case a "onPanelToggle" prop was provided.', () => {
        const onPanelToggle = () => null;
        const wrapper = shallow(<ToggablePanel {...props} onPanelToggle={onPanelToggle} isOpen/>);

        expect(wrapper.prop('isOpen')).toBe(true);
    });

    it('should propagate the "onPanelToggle" prop to the the "StatelessToggablePanel" component in case it was provided.', () => {
        const onPanelToggle = () => null;
        const wrapper = shallow(<ToggablePanel {...props} onPanelToggle={onPanelToggle}/>);

        expect(wrapper.prop('onPanelToggle')).toBe(onPanelToggle);
    });

    it('should propagate the "isOpen" state to the the "StatelessToggablePanel" component in case no "onPanelToggle" prop was provided.', () => {
        const wrapper = shallow(<ToggablePanel {...props}/>);

        expect(wrapper.prop('isOpen')).toBe(wrapper.state('isOpen'));
    });

    it('should propagate the "toggle" instance method to the the "StatelessToggablePanel" component in case no "onPanelToggle" prop was provided.', () => {
        const wrapper = shallow(<ToggablePanel {...props}/>);

        expect(wrapper.prop('onPanelToggle')).toBe(wrapper.instance().toggle);
    });
});

describe('<StatelessToggablePanel/>', () => {
    let props;

    beforeEach(() => {
        props = {
            theme: {
                'panel': 'baseClassName',
                'panel--isOpen': 'isOpenClassName',
                'panel__headline': 'panelHeadlineClassName',
                'panel__contents': 'panelContentsClassName'
            },
            onPanelToggle: () => null,
            children: 'Foo children',
            HeadlineComponent: createStubComponent('HeadlineComponent'),
            IconButtonComponent: createStubComponent('IconButtonComponent')
        };
    });

    it('should render correctly.', () => {
        const wrapper = shallow(<StatelessToggablePanel {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render both the "panel" and "panel--isOpen" className of the theme in case the "isOpen" prop is truthy.', () => {
        const wrapper = shallow(<StatelessToggablePanel {...props} isOpen/>);

        expect(wrapper.prop('className')).toContain(props.theme.panel);
        expect(wrapper.prop('className')).toContain(props.theme['panel--isOpen']);
    });
});

describe('<Header/>', () => {
    let context;
    let props;

    beforeEach(() => {
        context = {
            onPanelToggle: jest.fn()
        };
        props = {
            children: 'Foo children',
            isPanelOpen: false,
            theme: {
                'panel__headline': 'panelHeadlineClassName',
                'panel__headline--noPadding': 'panelHeadlineNoPaddingClassName',
                'panel__toggleBtn': 'panelToggleBtnClassName'
            },
            noPadding: false,
            HeadlineComponent: createStubComponent(),
            IconButtonComponent: createStubComponent(),
            openedIcon: 'opened-icon-prop',
            closedIcon: 'closed-icon-prop'
        };
    });

    it('should render correctly.', () => {
        const wrapper = shallow(<Header {...props}/>, {context});

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should propagate the "isPanelOpen" prop to the "aria-expanded" attribute.', () => {
        expect(shallow(<Header {...props} isPanelOpen/>, {context}).prop('aria-expanded')).toBeTruthy();
        expect(shallow(<Header {...props} isPanelOpen={false}/>, {context}).prop('aria-expanded')).toBeFalsy();
    });

    it('should propagate the "openedIcon" prop to the Icon component if the "isPanelOpen" prop is truthy.', () => {
        const wrapper = shallow(<Header {...props} isPanelOpen/>, {context});
        const icon = wrapper.find(props.IconButtonComponent);

        expect(icon.prop('icon')).toBe(props.openedIcon);
    });

    it('should propagate the "closedIcon" prop to the Icon component if the "isPanelOpen" prop is falsy.', () => {
        const wrapper = shallow(<Header {...props} isPanelOpen={false}/>, {context});
        const icon = wrapper.find(props.IconButtonComponent);

        expect(icon.prop('icon')).toBe(props.closedIcon);
    });
});

describe('<Contents/>', () => {
    let context;
    let props;

    beforeEach(() => {
        context = {
            onPanelToggle: jest.fn()
        };
        props = {
            children: 'Foo children',
            isPanelOpen: false,
            theme: {
                'panel__contents': 'panelContentsClassName',
                'panel__contents--noPadding': 'panelContentsNoPaddingClassName'
            },
            noPadding: false
        };
    });

    it('should render correctly.', () => {
        const wrapper = shallow(<Contents {...props}/>, {context});

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should negate the "isPanelOpen" prop to the "aria-hidden" attribute.', () => {
        expect(shallow(<Contents {...props} isPanelOpen/>, {context}).prop('aria-hidden')).toBeFalsy();
        expect(shallow(<Contents {...props} isPanelOpen={false}/>, {context}).prop('aria-hidden')).toBeFalsy();
    });
});
