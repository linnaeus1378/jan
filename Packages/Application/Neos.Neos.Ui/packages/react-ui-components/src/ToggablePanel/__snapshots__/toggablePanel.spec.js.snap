// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contents/> should render correctly. 1`] = `
<UnmountClosed
  isOpened={false}
  springConfig={
    Object {
      "damping": 30,
      "stiffness": 300,
    }
  }
>
  <div
    aria-hidden="true"
    className="panelContentsClassName"
  >
    Foo children
  </div>
</UnmountClosed>
`;

exports[`<Header/> should render correctly. 1`] = `
<div
  aria-expanded={false}
  onClick={[MockFunction]}
  role="button"
>
  <Component
    className="panelHeadlineClassName"
    type="h2"
  >
    Foo children
  </Component>
  <Component
    className="panelToggleBtnClassName"
    hoverStyle="clean"
    icon="closed-icon-prop"
  />
</div>
`;

exports[`<StatelessToggablePanel/> should render correctly. 1`] = `
<section
  className="baseClassName"
>
  Foo children
</section>
`;

exports[`<ToggablePanel/> should render correctly. 1`] = `
<StatelessToggablePanel
  HeadlineComponent={[Function]}
  IconButtonComponent={[Function]}
  isOpen={true}
  onPanelToggle={[Function]}
  theme={
    Object {
      "panel": "baseClassName",
      "panel--isOpen": "isOpenClassName",
      "panel__contents": "panelContentsClassName",
      "panel__headline": "panelHeadlineClassName",
    }
  }
>
  <Header
    HeadlineComponent={[Function]}
    IconButtonComponent={[Function]}
    closedIcon="chevron-circle-down"
    isPanelOpen={true}
    key="header"
    openedIcon="chevron-circle-up"
    theme={
      Object {
        "panel": "baseClassName",
        "panel--isOpen": "isOpenClassName",
        "panel__contents": "panelContentsClassName",
        "panel__headline": "panelHeadlineClassName",
      }
    }
  >
    My Header
  </Header>
  <Contents
    HeadlineComponent={[Function]}
    IconButtonComponent={[Function]}
    isPanelOpen={true}
    key="contents"
    theme={
      Object {
        "panel": "baseClassName",
        "panel--isOpen": "isOpenClassName",
        "panel__contents": "panelContentsClassName",
        "panel__headline": "panelHeadlineClassName",
      }
    }
  >
    My my Content
  </Contents>
</StatelessToggablePanel>
`;
